src/generate_trigger.ts

```diff
@@ -0,0 +1,120 @@
+import { parse, path } from "./deps.ts";
+
+export type WorkflowRecord = {
+  "callback_id": string;
+};
+
+interface Inputs {
+  [key: string]: {value: string};
+}
+
+export type TriggerFileRecord = {
+  "type": string;
+  "name": string;
+  "description"?: string;
+  "workflow": WorkflowRecord;
+  "inputs"?: Inputs;
+};
+
+export type TriggersPayload = {
+  triggers: TriggerFileRecord[];
+};
+
+export interface Response {
+  ok: boolean;
+  files: Array<string>;
+  error?: {
+    message: string;
+  } | null;
+}
+
+
+const generateTriggerFile = async (): Promise<Response> => {
+  // Parse JSON payload
+  const source = parse(Deno.args).source as string;
+  if (!source) throw new Error("A source path needs to be defined");
+  const payload: TriggersPayload = JSON.parse(source);
+
+  // Sort alphabetically
+  payload.triggers.sort((a, b) => a.name.localeCompare(b.name));
+
+  // Create the triggers directory if it does not exist
+  const directory = "triggers";
+  try {
+    await Deno.stat(directory);
+  } catch (error) {
+    if (error instanceof Deno.errors.NotFound) { // create directory
+      await Deno.mkdir(directory);
+    } else {
+      throw error;
+    }
+  }
+
+  // Generate the files
+  const files: Array<string> = [];
+  await Promise.all(payload.triggers.map(async (tr: TriggerFileRecord) => {
+    console.log(
```

Comment:
Many of the hooks use stdout to communicate with the CLI, making use of `console.log` troublesome (as the CLI will assume it will be part of the hook response). However, that doesn't seem to be an issue when I was testing this out. Does this hooks use a different mechanism to pass information back to the CLI? Or is it a one-way hook (that is, it doesn't expect any particular response from the SDK)?

Reply:
I think it would be the latter (not interacting with the SDK). Will remove this print statement anyway!